### PR TITLE
add more wallet methods

### PR DESCRIPTION
**Pull request by dashuser33**

### CODE REVIEW

File: src/lib/RPCDefinitions.ts [hunk 1]
```diff
@@ -528,3 +528,19 @@ export interface Transaction {
...
+/**
+ * Result of some estimate fee methods
+ */
+export interface FeeEstimation {
+  /**
+   * The estimated fee the transaction should pay in order to be included within the specified
+   * number of blocks. If the node doesn’t have enough information to make an estimate,
+   * the value -1 will be returned
+   */
+  readonly feerate: number;
```
Comment (jhodges10) [inline src/lib/RPCDefinitions.ts]: Thoughts on `feeRate` vs `feerate`?

I had to read it several times before I understood what it was supposed to be.

File: src/lib/darkcoinClientWallet.spec.ts [hunk 2]
```diff
@@ -91,6 +91,71 @@ test('send to address', async t => {
...
+});
+
+test('Estimate smart fee', async t => {
+  await client.estimateSmartFee(1).then(r => {
+    t.true(r.result.feerate > 0);
+    t.true(r.result.blocks > 0);
+  });
+});
+
+test('Dumped wallet should be imported without errors', async t => {
```
Comment (jhodges10) [inline src/lib/darkcoinClientWallet.spec.ts]: This one is clever. Nice work.

**Comment by jhodges10** [comment]
Other than `feerate` vs `feeRate`, this looks good.

**Review by jhodges10 — APPROVED** [review]

File: src/lib/RPCDefinitions.ts (hunk 1 above)
Comment (dashuser33) [reply src/lib/RPCDefinitions.ts]: @jhodges10 this is mapping the json object, according to [this](https://stackoverflow.com/a/2738375), json should be treated as case-sensitive.

**Comment by dashuser33** [comment]
Thanks for reviews. Merging.

File: src/lib/RPCDefinitions.ts (hunk 1 above)
Comment (jhodges10) [reply src/lib/RPCDefinitions.ts]: Got it! Man it's not easy to read...

Comment (dashuser33) [reply src/lib/RPCDefinitions.ts]: I know! the alternative would be for us to actually transform the inputs into something else. I like the idea, but it is going to require a lot more work.

**Pull request merged**